rancher2/resource_rancher2_service_account_token.go

```diff
@@ -0,0 +1,189 @@
+package rancher2
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	norman "github.com/rancher/norman/types"
+	managementClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+func resourceRancher2ServiceAccountToken() *schema.Resource {
```

Comment:
Same as above, update naming. My suggestion would be to call this resource `resourceRancher2UserAccountAPIToken` and this func `resourceRancher2UserAccountAPITokenCreate` and then propagate that naming to the rest of the code

Reply:
Do we want to keep the naming in line with the existing `rancher2_user` (functions `resourceRancher2User() / resourceRancher2UserCreate()`) and `rancher2_token` (functions `resourceRancher2Token() / resourceRancher2TokenCreate()`) resources?

If so, perhaps we can consider just simply naming it `rancher2_user_token` (function `resourceRancher2UserToken`). Otherwise we introduce the wording 'user' vs 'user account' and  'api token' vs 'token'.

Reply:
@sboschman I had a discussion with a few folks about the naming here, and I think it makes more sense to _split the resources_ you have implemented into separate `userAccount` and `token` resources because in Rancher, those resources are different things.

Then you can have the "login to userAccount" action and create/update/delete operations in different resource files. That makes more sense to me personally as well because I felt like we are logging into Rancher and creating a token.. for the same resource "Create" func which is confusing.

So let's have
`resourceRancher2UserAccount` and `resourceRancher2UserAccountToken`

Does this make sense?

Reply:
We only have 3 (terraform) resources, not 4 as I seem to understand from your comment.

`resourceRancher2User`: This existing resource allows user management, not just management of the provider user we use to connect to the Rancher API.
```
# Create a new rancher2 User
resource "rancher2_user" "foo" {
  name = "Foo user"
  username = "foo"
  password = "changeme"
  enabled = true
}
```

`resourceRancher2Token`: This existing resource only allows creating tokens for the provider user (the user we use to provision Rancher, which has super admin permissions)
```
# Create a new rancher2 Token
resource "rancher2_token" "foo" {
  description = "foo token"
  ttl = 1200
}
```

`resourceRancher2<something>Token`: This is a new resource which allows creating tokens for any Rancher User. It does not create the user itself. The user still has to be created with the `rancher2_user` resource. Which allows us to create a Rancher User with limited permission and add a token to this user. 

Unfortunately, at the time this PR was made, the Rancher API did not offer any endpoint to create tokens for a user other than the user the token used to access the Rancher API belongs to.
The login flow you mention is a workaround. The only way to acquire a Rancher API token is to log in using username+password, unless you already have a token for the user of course.

So, from a terraform provider point of view, the only resource we want to expose is a single resource which can be used to create tokens for any existing Rancher user.

Internally the terraform provider has to do a trick to make this happen:
1. Use the username+password of the user we want to create tokens for to login to Rancher and acquire a Rancher API token bound to this user
1. Make a call to the Rancher API Token endpoint authenticating with the token from step 1 to create an additional token, this is the token we are returning as 'the token' for the terraform resource. 
1. Logout the temporarily token we acquired in step 1.

It clearly is a different "flow" of things compared to all the other resources, which all can simply use the provider token to authenticate to the Rancher API. The `doUserLogin`, `doUserLogout`, etc are just helper functions very specific to this new resource. We do not return the temporary token acquired in step 1 as it has a limited lifespan, controlled by Rancher, and we want the option to create more than one token.

I am happy to name the new resource `resourceRancher2CustomUserToken` though, even if I don't really understand the reasoning behind the 'custom' part.

Usage would look like this:
```hcl
# Create a rancher2 Token
resource "rancher2_user" "foo" {
  name = "foo"
  username = "foo"
  password = "changeme"
  enabled = true
}
resource "rancher2_global_role_binding" "foo-login" {
  name = "foo-login-binding"
  global_role_id = "user-base"
  user_id = rancher2_user.foo.id
}
resource "rancher2_custom_user_token" "foo" {      <<< what does this 'custom' mean?
  username = rancher2_user.foo.username
  password = rancher2_user.foo.password
  description = "foo token"
  ttl = 0
  depends_on = [
    rancher2_global_role_binding.foo-login
  ]
}
```

Reply:
@sboschman Thank you for the clarification here.

Yes, please rename the resource to be `resourceRancher2CustomUserToken` so that it's clear this resource is only to generate a new Rancher API token for a new user (a custom user not a token for the default / admin user, that is why I think custom is good wording).